### DISCUSSION TITLE
Custom Drag/Drop Multiple Images Hanging...

### DISCUSSION THREAD

**Discussion by Chipperchap** [sircharlo/meeting-media-manager#757] [Q&A] [answered]
Sometimes if multiple images are copied over at once M3 seems to hang. When M3 is closed and reopened the images are generally all there though.
Video (a bit boring) [HERE](https://capture.dropbox.com/FNm4jRQCKu2IiufH)

It doesn't seem to occur with JWPUB files to the same extent.

It does eventually come right, but just seems to take a long time... or am I just being impatient? (Sorry).

**Comment by mtdvlpr** [comment, marked as answer]
Hmm, it's probably not hanging, but trying to upload each image to your congregation server, one at a time. There should be a progress bar for each upload, but it seems there is a bug with the package we're using: https://github.com/perry-mitchell/webdav-client/issues/319.

I will try to optimize this so that files are uploaded simultaneously and then I can create my own progress and show that.

**Comment by mtdvlpr** [comment]
Implemented in #758 :)

**Comment by Chipperchap** [comment]
Great wonderful! Thank you so much. 🏆

**Reply by mtdvlpr** [reply]
@Chipperchap, does the latest release indeed solve your issue or is it still hanging?

**Reply by Chipperchap** [reply]
Love the progress bar. Much helpful and muchly appreciated. 

I have had a play again and I think part of the problem may have been an error(s) on my part:
1. Impatience 
2. Trying to upload a couple of .bmp pictures which I had tucked in among some .png pictures. It uploads .bmp files but seems to take some time ... and does not make them available for display which temporarily confused me (I am easily confused). 😕

So in summary - everything is great! Well done and thank you so much!

(By the way we don't need provision to display .bmp files, it was just part of a territory map assignment I was working on.) 😁
